plugins/tableselection/plugin.js

```diff
@@ -811,9 +812,12 @@
 		if ( !selectedCells.length ||
 			// It's single range that does not fully contain table element and is not boundary, e.g. collapsed selection within
 			// cell, part of cell etc.
-			( selectedCells.length === 1 && !rangeContainsTableElement( selection.getRanges()[ 0 ] )  && !boundarySelection ) ||
+			( selectedCells.length === 1 && !rangeContainsTableElement( selection.getRanges()[ 0 ] ) && !boundarySelection ) ||
 			// It's a boundary position but with no table pasted.
-			( boundarySelection && !pastedTable ) ) {
+			( boundarySelection && !pastedTable ) ||
+			// There is some selected content before table (#875).
+			( selectionAncestor && selectionAncestor.$.nodeType == 1 && selectionAncestor.hasClass( 'cke_editable' ) )
+			) {
```

Comment:
This `if` statement became too complicated and unreadable, please extract or simplify it. You can also combine this statement with check: https://github.com/ckeditor/ckeditor4/blob/ad381679b4436f489a65f6f397680da338024b0f/plugins/tableselection/plugin.js#L784-L786
and create a function like: `useTableSelection( selection )`
so entire code will  be reduce to:
```
if ( !useTableSelection( selection ) ) {
    return;
}
```
In `useTableSelection( selection )` you can also use the return early pattern to prevent unecessary checks.